### PR TITLE
[projects] allow only github.com and gitlab.com

### PR DESCRIPTION
**Pull request by AlexTugarev**

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **alextugarev** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/projects/NewProject.tsx
Comment (gtsiolis) [inline components/dashboard/src/projects/NewProject.tsx:345]: **question**: Does this limit Projects for repositories only from `github.com` and `gitlab.com`? I thought Projects could support repositories from self hosted GitHub and GitLab instance, but this is also something we could leave out of the scope of this first iteration. 💭

Comment (AlexTugarev) [reply components/dashboard/src/projects/NewProject.tsx:345]: 1. The current GitHub App integration is hard coded with github.com. 
2. I'm new to the GitLab integration for prebuilds in general. It would require testing as well.

Comment (gtsiolis) [reply components/dashboard/src/projects/NewProject.tsx:345]: Aha! Thanks for the insight, @AlexTugarev! Sounds good to leave this out of the scope for now until further demand. Added https://github.com/gitpod-io/gitpod/issues/5115 to track this feature request.

Comment (gtsiolis) [reply components/dashboard/src/projects/NewProject.tsx:345]: This would still need an approval from a [dashboard owner](https://github.com/gitpod-io/gitpod/pull/5113#issuecomment-895112689). Looping in @jankeromnes and @svenefftinge in case any of them can take a look at this two-line change. 🏓

Comment (jankeromnes) [reply components/dashboard/src/projects/NewProject.tsx:345]: Fixing this in https://github.com/gitpod-io/gitpod/pull/5192 😁 also taking a quick look!

**Review by jankeromnes — APPROVED** [review]
Many thanks! The code seems to do what's intended, but I don't understand _why_ this change is needed.

Could you please elaborate?

Also, @gtsiolis makes a great point about self-managed GitLabs. I'm aware that we haven't tested them and they might break, but maybe it's still preferable to have something that might "accidentally work" for self-managed GitLabs, rather than something that "definitely fails" (i.e. it's filtered out)?

Still, approving in case you feel this change is really necessary. 😊 Maybe I'm missing something.

/hold

/lgtm

/approve no-issue

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 6549ee27944cd231da149ac74d1af4c06c63c5e0</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5113#pullrequestreview-729630157" title="Approved">jankeromnes</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/5113#pullrequestreview-729630157" title="Approved">jankeromnes</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by AlexTugarev** [comment]
> Also, @gtsiolis makes a great point about self-managed GitLabs. I'm aware that we haven't tested them and they might break, but maybe it's still preferable to have something that might "accidentally work" for self-managed GitLabs, rather than something that "definitely fails" (i.e. it's filtered out)?

@jankeromnes, I'm not sure about the chances that it work nicely out-of-the box with self-manages GitLabs. It sure does not with GitHub Enterprise. But in general, we should deliver tested feature only.

**Pull request merged**
